### ISSUE TITLE
Functions that use the precompiled DLL in non-root folder aren't returned by "api/functions" call

### ISSUE THREAD

**Issue by alrod** [projectkudu/kudu#2334]
I have the following structure:

```
FunctionName\
  - function.json
bin
  - assembly.dll
```

The function FunctionName doesn't show up at all, until I add an empty content file to the folder. Expected behavior: show only function.json if a script content file doesn't exist.

move from portal repo:
https://github.com/projectkudu/AzureFunctionsPortal/issues/925

**Comment by watashiSHUN** [comment]
so what should the `script URL` look like in this case? in my simple fix, `api/functions` simply return `null` which will make portal display nothing (change it to the address of `assembly.dll` is not going to generate a pretty display either?)

```
[{
	"name": "dllFunTest",
	"function_app_id": null,
	"script_root_path_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/site/wwwroot/dllFunTest/",
	"script_href": null,
	"config_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/site/wwwroot/dllFunTest/function.json",
	"secrets_file_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/data/functions/secrets/dllFunTest.json",
	"href": "https://shunfunctionapp.scm.azurewebsites.net/api/functions/dllFunTest",
	"config": {
		"bindings": [{
			"type": "manualTrigger",
			"direction": "in",
			"name": "input"
		}],
		"disabled": false
	},
	"files": null,
	"test_data": ""
},
{
	"name": "ManualTriggerCSharp1",
	"function_app_id": null,
	"script_root_path_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/site/wwwroot/ManualTriggerCSharp1/",
	"script_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/site/wwwroot/ManualTriggerCSharp1/run.csx",
	"config_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/site/wwwroot/ManualTriggerCSharp1/function.json",
	"secrets_file_href": "https://shunfunctionapp.scm.azurewebsites.net/api/vfs/data/functions/secrets/ManualTriggerCSharp1.json",
	"href": "https://shunfunctionapp.scm.azurewebsites.net/api/functions/ManualTriggerCSharp1",
	"config": {
		"bindings": [{
			"type": "manualTrigger",
			"direction": "in",
			"name": "input"
		}],
		"disabled": false
	},
	"files": null,
	"test_data": "Test Value"
}]
```

**Comment by lindydonna** [comment]
@watashiSHUN Thanks for the fix!  Do you know when it will be deployed?

**Comment by ahmelsayed** [comment]
@watashiSHUN this should be closed by #2338, right?
